impl/MainApp.cpp

```diff
@@ -46,10 +46,15 @@ constexpr std::uint32_t FRAMES_IN_FLIGHT = 2;
 vk_gltf_viewer::MainApp::MainApp() {
     const vulkan::pipeline::BrdfmapComputer brdfmapComputer { gpu.device };
 
-    const vk::raii::DescriptorPool descriptorPool {
-        gpu.device,
-        brdfmapComputer.descriptorSetLayout.getPoolSize().getDescriptorPoolCreateInfo(),
-    };
+
+    vk::DescriptorPoolCreateInfo descriptorPoolCreateInfo = brdfmapComputer.descriptorSetLayout.getPoolSize().getDescriptorPoolCreateInfo();
```

Comment:
This would not be work. `vku::PoolSizes::getDescriptorPoolCreateInfo` returns both `vk::DescriptorPoolCreateInfo` and `std::vector<vk::DescriptorPoolSize>` via `vku::RefHolder` (former references the latter), but store it into `vk::DescriptorPoolCreateInfo` will destroy the pool size vector.

You should pass `vk::DescriptorPoolCreateFlagBits::eFreeDescriptorSet` to the method by parameter and use the struct directly.

Reply:
ok thanks :)